src/saltext/vault/states/vault_pki.py

```diff
@@ -0,0 +1,502 @@
+
+    Required policy:
+
+    .. code-block:: vaultpolicy
+
+            path "{mount}/issuer/{issuer_ref}/sign/{role_name}" {
+                capabilities = ["update"]
+            }
+            # in case of sign_verbatim
+            path "{mount}/issuer/{issuer_ref}/sign-verbatim/{role_name}" {
+                capabilities = ["update"]
+            }
+
+    name
+        Path to the certificate file.
+
+    common_name
+        Common name to be set for the certificate.
+
+    role_name
+        PKI role to be used for issuing the certificate from Vault.
+
+    private_key
+        Path or PEM formatted text of the private key used to sign CSR for the certificate.
+
+    mount
+        The mount path the PKI backend is mounted to. Defaults to ``pki``.
+
+    ttl
+        Specifies the Time To Live value to be used for the validity period of the requested certificate,
+        provided as a string duration with time suffix. Hour is the largest suffix. Defaults to ``720h`` or 30 days.
+
+    ttl_remaining
+        Specifies the Time To Live value to be used for checking remaining period before expiration
+        after which certificate should be renewed.
+        Provided as a string duration with time suffix. Hour is the largest suffix. Defaults to ``168h`` or 7 days.
+
+    issuer_ref
+        Override role's issuer for the certificate. Defaults to the one specified in the role.
+
+    encoding
+        Encoding to be used for the certificate file. Valid options are ``pem``, ``pkcs7_pem``, ``der``, ``pkcs7_der``. Defaults to ``pem``.
+
+    append_ca_chain
+        If set to true will append CA chain to the certificate. Defaults to ``false``.
+
+        .. note::
+            This will append all CA certificates except self-signed (as they shouldn't be in the chain anyway)!
+
+    sign_verbatim
+        If set to true, the resulting certificate follows the CSR exactly.
+        Otherwise, only ``CN`` can be set for the subject, any other subject parameter (like ``O``) is ignored.
+
+        .. warning::
+            This option is using a potentially dangerous endpoint. Be careful when using that option, as roles
+            are not restricting what can be issued anymore.
+
+    private_key_passphrase
+        Password for the private key if encrypted.
+
+    reissue
+        Always reissue the certificate. Defaults to ``false``.
+
+    kwargs
+        Any other parameter accepted by ``file_managed`` execution module or Vault PKI
+        :obj:`sign_certificate <saltext.vault.modules.vault_pki.sign_certificate>` execution module.
+    """
+
+    if encoding not in ["der", "pem", "pkcs7_der", "pkcs7_pem"]:
+        raise CommandExecutionError(
+            f"Invalid value '{encoding}' for encoding. Valid: der, pem, pkcs7_der, pkcs7_pem"
+        )
+
+    if timestring_map(ttl_remaining, cast=int) >= timestring_map(ttl, cast=int):
+        raise CommandExecutionError("The ttl_remaning cannot be larger or equal to ttl.")
```

Comment:
Sorry, missed this. Anything that raises exceptions should be moved into the `try...except` block below, otherwise we're hard-crashing. Also I think `SaltInvocationError` would fit better here, but doesn't really make a difference once exceptions are caught.